scripts/ci/before_install.sh

```diff
@@ -51,12 +51,18 @@ elif [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
   ln -s -f /usr/local/bin/pip3 /usr/local/bin/pip
 fi
 
+# fetch tags for `git-describe`, since
+# - can't rely on $TRAVIS_TAG for `edge` (master) releases, and
+# - `snapcraft` also uses `git-describe` for version detection
+git fetch --tags
+TAG_MAJOR="$(git describe --tags | sed -r 's/^v?([0-9]+)\.[0-9]+\.[0-9]+.*/\1/')"
+[[ -n "$TAG_MAJOR" ]] || exit 1  # failed to detect major version
 if [[ -n "$TRAVIS_TAG" ]]; then
   if [[ $(echo "$TRAVIS_TAG" | grep -E '^[0-9]+\.[0-9]+\.[0-9]+$') ]]; then
-    echo "export SNAP_CHANNEL=stable" >>env.sh
+    echo "export SNAP_CHANNEL=stable,v$TAG_MAJOR/stable" >>env.sh
   else
-    echo "export SNAP_CHANNEL=beta" >>env.sh
+    echo "export SNAP_CHANNEL=beta,v$TAG_MAJOR/beta" >>env.sh
   fi
 else
-  echo "export SNAP_CHANNEL=edge" >>env.sh
+  echo "export SNAP_CHANNEL=edge,v$TAG_MAJOR/edge" >>env.sh
 fi
```

Comment:
Oops, missed that this is only for 0.94-dev branch. I guess we need to port to the master too.

Reply:
yes, will do later